### PR TITLE
Fixed SYSEX Hauptwerk max. length in MIDI event editor https://github.com/GrandOrgue/grandorgue/issues/1207

### PR DESCRIPTION
**Pull request by willeke1234**
Resolves: #1207

Changed LengthLimit of MIDI_S_HW_NAME_STRING and MIDI_S_HW_STRING from 15 to 16 and LengthLimit of MIDI_S_HW_NAME_LCD and MIDI_S_HW_LCD from 31 to 32

### CODE REVIEW

**Comment by oleg68** [comment]
@willeke1234 Please register your change in the the CHANGELOG.md file.

**Review by oleg68 — APPROVED** [review]

File: src/grandorgue/midi/GOMidiSender.cpp [hunk 1]
```diff
@@ -294,9 +294,9 @@ unsigned GOMidiSender::StartLimit(GOMidiSenderMessageType type) {
 
 unsigned GOMidiSender::LengthLimit(GOMidiSenderMessageType type) {
   if (type == MIDI_S_HW_NAME_STRING || type == MIDI_S_HW_STRING)
-    return 15;
+    return 16;
```
[the remark this answers is not in the document]
Comment (rousseldenis) [reply src/grandorgue/midi/GOMidiSender.cpp:297]: @oleg68 Mmmh, LenghtLimit is used there:

image


https://github.com/GrandOrgue/grandorgue/blob/master/src/grandorgue/dialogs/midi-event/GOMidiEventSendTab.cpp#L309


And if I understand well, SetRange() will do a range from 0->16, so a size of 17. Am I wrong ?

[the remark this answers is not in the document]
Comment (oleg68) [reply src/grandorgue/midi/GOMidiSender.cpp:297]: > And if I understand well, SetRange() will do a range from 0->16, so a size of 17. Am I wrong ?

It is the range of possible value for the "Length" field in the Midi Event Editor. You can right-click on a midi element, go to the `Send` tab, select any SYSEX event and try to enter the value in the Length field. Now GO does not allow to enter 32 or 16 here.

[the remark this answers is not in the document]
Comment (larspalo) [reply src/grandorgue/midi/GOMidiSender.cpp:297]: @rousseldenis The way the length value is used here it's not a problem, it's not an array where index 0 actually would be the first position (and if only that position was set, would indicate a string length of 1). That said, I don't have the LCD hardware to be able to test the functionality in reality.

**Review by larspalo — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**